ios/UtilsPlugin.mm

```diff
@@ -307,67 +307,69 @@ - (void)shareText:(NSDictionary *)jsonObject {
-			fileExistsAtPath:@"/etc/apt"]) {
-		jailBroken = YES;
-	}
-
-	NSError *error;
-	NSString *stringToBeWritten = @"This is a test.";
-	[stringToBeWritten writeToFile:@"/private/jailbreak.txt" atomically:YES
-		encoding:NSUTF8StringEncoding error:&error];
-	if(error == nil) {
-		//Device is jailbroken
-		jailBroken = YES;
-	} else {
-		[[NSFileManager defaultManager] removeItemAtPath:@"/private/jailbreak.txt" error:nil];
-	}
-
-	if([[UIApplication sharedApplication]
-			canOpenURL:[NSURL URLWithString:@"cydia://package/com.example.package"]]) {
+  BOOL jailBroken = NO;
+  #if !(TARGET_IPHONE_SIMULATOR)
+  if ([[NSFileManager defaultManager]
+      fileExistsAtPath:@"/Applications/Cydia.app"]) {
+    jailBroken = YES;
+  } else if([[NSFileManager defaultManager]
+		  fileExistsAtPath:@"/Library/MobileSubstrate/MobileSubstrate.dylib"]) {
+    jailBroken = YES;
+  } else if([[NSFileManager defaultManager]
+      fileExistsAtPath:@"/bin/bash"]) {
+    jailBroken = YES;
+  } else if([[NSFileManager defaultManager]
+      fileExistsAtPath:@"/usr/sbin/sshd"]) {
+    jailBroken = YES;
+  } else if([[NSFileManager defaultManager]
+      fileExistsAtPath:@"/etc/apt"]) {
+    jailBroken = YES;
+  }
+
+  NSError *error;
+  NSString *stringToBeWritten = @"This is a test.";
+  [stringToBeWritten writeToFile:@"/private/jailbreak.txt" atomically:YES
+    encoding:NSUTF8StringEncoding error:&error];
+  if(error == nil) {
+    //Device is jailbroken
+    jailBroken = YES;
+  } else {
+    [[NSFileManager defaultManager] removeItemAtPath:@"/private/jailbreak.txt" error:nil];
+  }
+
+  if([[UIApplication sharedApplication]
+      canOpenURL:[NSURL URLWithString:@"cydia://package/com.example.package"]]) {
 		//Device is jailbroken
-		jailBroken = YES;
-	}
-#endif
-
-	NSString *ret = (jailBroken)? @"true": @"false";
-	[[PluginManager get] dispatchJSEvent:[NSDictionary dictionaryWithObjectsAndKeys:
-		@"utilsJailBroken",@"name",
-		ret,@"jb",
-		nil]];
+    jailBroken = YES;
+  }
+  #endif
+
+  NSString *ret = (jailBroken)? @"true": @"false";
+  [[PluginManager get] dispatchJSEvent:[NSDictionary dictionaryWithObjectsAndKeys:
+    @"utilsJailBroken",@"name",
+    ret,@"jb",
+    nil]];
 }
 
 - (void)getAdvertisingId:(NSDictionary *)jsonObject {
-	NSString *limit_tracking = @"1";
-	NSString *id = [[ASIdentifierManager sharedManager].advertisingIdentifier UUIDString];
-
-	if([[ASIdentifierManager sharedManager] isAdvertisingTrackingEnabled]) {
-		limit_tracking = @"0";
-	}
-	[[PluginManager get] dispatchJSEvent:[NSDictionary dictionaryWithObjectsAndKeys:
-		@"utilsAdvertisingId",@"name",
-		id,@"id",
-		limit_tracking, @"limit_tracking",
-		nil]];
+  NSString *limit_tracking = @"1";
+  NSString *id = [[ASIdentifierManager sharedManager].advertisingIdentifier UUIDString];
+ 
+  if([[ASIdentifierManager sharedManager] isAdvertisingTrackingEnabled]) {
+    limit_tracking = @"0";
+  }
+  else if([id isEqualToString :@"00000000-0000-0000-0000-000000000000"]){
```

Comment:
else if is enough? 

if there is possibility of isAdvertisingTrackingEnabled is true and adv id is 00000000-0000-0000-0000-000000000000,
it will break. @vivekb91